### DISCUSSION TITLE
What is reason for output.charset to be `ascii` as default in RSBuild ?

### DISCUSSION THREAD

**Discussion by Verthon** [web-infra-dev/rsbuild#3244] [Q&A] [answered]
Hello, I have noticed that RSBuild sets output.charset to **ASCII** by default, whereas **SWC** sets it to **UTF-8**. 
Do you know why RSBuild favors **ASCII** over **UTF-8**? Thanks.

**Comment by chenjiahan** [comment]
I think this is a historical reason, the early implementation of Rsbuild served an internal framework that used `ascii` as the default value.

I will re-evaluate this default value before Rsbuild 1.0 is released.

**Comment by chenjiahan** [comment, marked as answer]
The default charset changed to `utf8`: https://github.com/web-infra-dev/rsbuild/pull/3250

**Reply by Verthon** [reply]
Hello, @chenjiahan do you think it is worth mentioning this change in the Breaking changes page https://github.com/web-infra-dev/rsbuild/discussions/2508 ?

**Reply by chenjiahan** [reply]
Done 😄

**Comment by felixmosh** [comment]
Looks like it breaks some part of lodash, as object with none ascii properties.
![image](https://github.com/user-attachments/assets/411415e2-3cba-45a7-bb5b-6f224b604b18)

**Reply by chenjiahan** [reply]
can you provide an reproduction?

**Reply by felixmosh** [reply]
I've checked it, looks like it is a fault of my Nginx.
I will change the output to ascii on my project.
